plugins/processors/t128_transform/t128_transform_state_change_test.go

```diff
@@ -74,182 +74,254 @@ func TestStateChangeLeavesUnmarkedFieldsInTact(t *testing.T) {
+				{},
+				{"/state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "leave-matching-existing-same",
+			Fields: map[string]string{"/state": "/state"},
+			Samples: []sample{
+				{"/state": 45},
+				{"/state": 45},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{},
+		},
+		{
+			Name:   "leave-matching-existing-changed",
+			Fields: map[string]string{"/state": "/state"},
+			Samples: []sample{
+				{"/state": 45},
+				{"/state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "leave-matching-expired",
+			Fields: map[string]string{"/state": "/state"},
+			Samples: []sample{
+				{"/state": 45},
+				{"/state": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "leave-mismatching-new",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{},
+				{"/non-state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(50), "/state": int64(50)},
+		},
+		{
+			Name:   "leave-mismatching-existing-same",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 45},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(45)},
+		},
+		{
+			Name:   "leave-mismatching-existing-changed",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(50), "/state": int64(50)},
+		},
+		{
+			Name:   "leave-mismatching-expired",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(50), "/state": int64(50)},
+		},
+
+		{
+			Name:           "previous-new",
+			Fields:         map[string]string{"/state": "/state"},
+			PreviousFields: map[string]string{"/previous": "/state"},
+			Samples: []sample{
+				{"/state": "s1"},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: true,
+			Result:         sample{"/state": "s1"},
+		},
+		{
+			Name:           "previous-with-previous-value",
+			Fields:         map[string]string{"/state": "/state"},
+			PreviousFields: map[string]string{"/previous": "/state"},
+			Samples: []sample{
+				{"/state": "s1"},
+				{"/state": "s2"},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: true,
+			Result:         sample{"/state": "s2", "/previous": "s1"},
```

Comment:
Does this rely completely on the current and previous sample for the "previous" case? In other words if `current == previous` we would not produce anything but do we record that? I would imagine only the observed "previous" to be reported - but just want to confirm that behavior.

Reply:
That's correct. But your question helped me identify a bug in a codepath I hadn't tested. When the transform should produce nothing, it returns a `nil` value. That's currently being recorded as the previous which isn't correct in some cases.
[] test the new corner case